src/System.Management.Automation/engine/PSClassSearcher.cs

```diff
@@ -297,7 +297,7 @@ private PSClassInfo ConvertToClassInfo(PSModuleInfo module, ScriptBlockAst ast,
             foreach (var member in statement.Members)
             {
                 PropertyMemberAst propAst = member as PropertyMemberAst;
-                if (propAst != null)
+                if (propAst != null && (propAst.PropertyAttributes & PropertyAttributes.Hidden) != PropertyAttributes.Hidden)
```

Comment:
Perhaps use 'is' :  `if (member is PropertyMemberAst propAst && (propAst.PropertyAttributes & PropertyAttributes.Hidden) != PropertyAttributes.Hidden)`